CMakeVOL.cmake

```diff
@@ -39,6 +39,13 @@ if (HDF5_VOL_ALLOW_EXTERNAL)
   set (HDF5_FOUND 1)
   set (HDF5_LIBRARIES "${HDF5_LIBSH_TARGET};${LINK_LIBS};${LINK_COMP_LIBS};$<$<BOOL:${HDF5_ENABLE_PARALLEL}>:MPI::MPI_C>")
   set (HDF5_INCLUDE_DIRS "${HDF5_SRC_INCLUDE_DIRS};${HDF5_SRC_BINARY_DIR};$<$<BOOL:${HDF5_ENABLE_PARALLEL}>:${MPI_C_INCLUDE_DIRS}>")
+  set (HDF5_DIR "${HDF5_SOURCE_DIR}")
```

Comment:
I believe `HDF5_DIR` is supposed to point to an installed HDF5 directory location, so setting it to the source dir for HDF5 might cause issues. Might want to set it to something like `${CMAKE_INSTALL_PREFIX}`

Reply:
Do the VOLs actually use this HDF5_DIR variable - because I think it is only informational and shouldn't be used otherwise.

Reply:
It was used in `vol-tests`, which is now deprecated. I replaced it in the REST VOL's CMake with a direct reference to `HDF5_SOURCE_DIR`, and removed this line.

Reply:
`HDF5_SOURCE_DIR` comes from the HDF5 project and so isn't likely to be a variable when building the REST VOL, except for possibly in the case of the prebuilt HDF5 notion that we're looking at removing now

Reply:
If the VOLs are built as a subdirectory under the HDF5 library, then I think they should inherit HDF5's CMake variables, including this one.

Reply:
But it doesn't exist yet, until packaged?

Reply:
At least I think so.  How does the VOL actually use that variable?

Reply:
The VOL uses `HDF5_SOURCE_DIR` to locate the headers to include when the high level library is enabled (Specifically, to add `${HDF5_SOURCE_DIR}/hl/src` as an include directory).

I might be confused - I think `HDF5_SOURCE_DIR` is automatically created when `project(HDF5)` is invoked in HDF5's top level CMakeLists.txt, and because we're reworking the VOLs to be built as a CMake subdirectory of the HDF5 project, the CMakeLists.txt for the VOLs will have access to `HDF5_SOURCE_DIR`.